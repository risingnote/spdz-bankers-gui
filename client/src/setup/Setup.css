.Setup-panel {
  padding: 10px;
  padding-top: 0px;
  border: 2px solid lightblue;
  width: 30%;
  margin: 0 auto;
}
