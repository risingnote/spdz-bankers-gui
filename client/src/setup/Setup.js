/**
 * Responsible for displaying setup information based on props passed in.
 */
import React from 'react';
import './Setup.css'

const Setup = (props) => {
  const proxyServers = (proxyServerList) => {
    return proxyServerList.map( (proxyServer) => {
      return <p key={proxyServer.key}>{proxyServer.url}</p>
    })
  }

  return (
    <div>
        <button onClick={props.setupForRun}>
          Setup for SPDZ
        </button>
        <div className='Setup-status'>
          <h4>Spdz Proxy Servers</h4>
          {proxyServers(props.spdzProxyServerList)}
        </div>
    </div>
  )
} 

Setup.proptypes = {
  setupForRun: React.PropTypes.func.isRequired,
  spdzProxyServerList: React.PropTypes.arrayOf(React.PropTypes.object).isRequired
}

export default Setup