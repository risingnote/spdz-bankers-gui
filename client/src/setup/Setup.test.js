import React from 'react'
import Setup from './Setup'
import renderer from 'react-test-renderer'

describe('Setup display component rendering', () => {
  it('Renders as expected (compared to a snapshot) when passed properties', () => {
    const spdzProxyServerList = [
      {
      "key": "1",
      "url": "http://spdzProxy.one:4000",
      "status": "notconnected"
      },
      {
      "key": "2",        
      "url": "http://spdzProxy.two:4000",
      "status": "notconnected"
      }
    ]
    const tree = renderer.create(
      <Setup setupForRun={() => {}} spdzProxyServerList={spdzProxyServerList} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('Executes the call back when the setup button is clicked', () => {
    const mockCallBack = jest.fn((e) => {})

    const tree = renderer.create(
      <Setup setupForRun={mockCallBack} spdzProxyServerList={[]} />
    ).toJSON()

    let button = tree.children.find((element) => {return (element['type'] === 'button')})

    button.props.onClick()
    expect(mockCallBack).toHaveBeenCalled()
  })
})
