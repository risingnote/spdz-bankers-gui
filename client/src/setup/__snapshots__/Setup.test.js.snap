exports[`Setup display component rendering Renders as expected (compared to a snapshot) when passed properties 1`] = `
<div>
  <button
    onClick={[Function]}>
    Setup for SPDZ
  </button>
  <div
    className="Setup-status">
    <h4>
      Spdz Proxy Servers
    </h4>
    <p>
      http://spdzProxy.one:4000
    </p>
    <p>
      http://spdzProxy.two:4000
    </p>
  </div>
</div>
`;
